### PR TITLE
Support bundling .node files in ESM & CJS when targeting Node.js

### PR DESCRIPTION
**Pull request by Jarred-Sumner**
### What does this PR do?

image

Fixes https://github.com/oven-sh/bun/issues/13198

### How did you verify your code works?

Tests

### CODE REVIEW

**Comment by robobun** [comment]
@Jarred-Sumner, your commit 903c88ef24fdf375a9666a7e431eb553a4038b2e is building: [`#4225`](https://buildkite.com/bun/bun/builds/4225)

File: src/bundler/bundle_v2.zig [hunk 1]
```diff
@@ -5221,6 +5205,21 @@ pub const LinkerContext = struct {
...
+                // If this is a .napi addon and it's not node, we need to generate a require() call to the runtime
+                if (expr.data == .e_call and expr.data.e_call.target.data == .e_require_call_target and
+                    // if it's commonjs, use require()
+                    this.options.output_format != .cjs and
+                    // if it's esm and bun, use import.meta.require(). the code for __require is not injected into the bundle.
+                    !this.options.target.isBun())
+                {
+                    this.graph.generateRuntimeSymbolImportAndUse(
+                        source_index,
+                        Index.part(1),
```
Comment (paperclover) [inline src/bundler/bundle_v2.zig:5218]: part 1?

File: src/bundler/bundle_v2.zig [hunk 2]
```diff
@@ -5221,6 +5205,21 @@ pub const LinkerContext = struct {
...
+                        "__require",
+                        1,
+                    ) catch unreachable;
```
Comment (paperclover) [inline src/bundler/bundle_v2.zig]: can we avoid increasing the unreachable debt?

**Review by paperclover — CHANGES_REQUESTED** [review]

File: src/bundler/bundle_v2.zig (hunk 2 above)
Comment (Jarred-Sumner) [reply src/bundler/bundle_v2.zig]:
```suggestion
                    ) catch {};
```

**Pull request merged**